server/server.js

```diff
@@ -13,76 +13,163 @@
+			const server = app.listen(
+				PORT,
+				e => {
+					handleError(e);
+
+					logToTerminal(outputColor(`Listening on port ${PORT} `));
+
+					global.host = server.address().address;
+					global.port = server.address().port;
+				});
+
+			app.use(compression());
+			// Sample server root set to "/" -- must align with paths throughout
+			app.use("/", express.static(rootDir, options));
+			// Open up the Finsemble Components,services, and clients
+			app.use("/Finsemble", express.static(moduleDirectory, options));
+			// For Assimilation
+			app.use("/hosted", express.static(path.join(__dirname, "..", "hosted"), options));
+
+			// configs/openfin/manifest-local.json and configs/other/server-environment-startup.json
+			// Make the config public
+			app.use("/configs", express.static("./configs", options));
 
 			/**
-			 * Method called after the server has started.
-			 *
-			 * @param {function} done Function can take one argument; an error message if one occurred.
-			 *
-			 * @example
-			 * const post => {
-			 * 	try {
-			 * 		// do something that could throw an error
-			 *  } catch(e) {
-			 *		done(e);
-			 *  }
-			 * }
+			 * Fill in the ProxyMap with entries in order to proxy remote components.
+			 * For instance /connect4/blah/mycomponent.html would be proxied to http://connect4.chartiq.com/blah/mycomponent.html
+			 * 
+			 * This proxy further uses the referer http header to redirect assets from within the component.
 			 */
-			post: done => { done(); },
+			let ProxyMap = [
+				/* Examples
+					{
+						root: "connect4",
+						proxyTarget: "http://connect4.chartiq.com",
+						host: "connect4.chartiq.com"
+					},
+					{
+						root: "connect3",
+						proxyTarget: "http://connect3.chartiq.com",
+						host: "connect3.chartiq.com"
+					},
+					{
+						root: "connect2",
+						proxyTarget: "http://connect2.chartiq.com",
+						host: "connect2.chartiq.com"
+					}
+				*/
+			];
 
-			/**
-			 * Method called to update the server.
-			 *
-			 * @param {express} app The express server.
-			 * @param {function} cb The function to call once finished adding functionality to the server.
-			 * @example
-			 * const cb => {
-			 * 	try {
-			 * 		// do something that could throw an error
-			 *  } catch(e) {
-			 *		done(e);
-			 *  }
-			 * }*/
-			updateServer: (app, cb) => {
-				app.use(compression());
-				// Sample server root set to "/" -- must align with paths throughout
-				app.use("/", express.static(rootDir, options));
-				// Open up the Finsemble Components,services, and clients
-				app.use("/Finsemble", express.static(moduleDirectory, options));
-				// For Assimilation
-				app.use("/hosted", express.static(path.join(__dirname, "..", "hosted"), options));
-
-				// configs/openfin/manifest-local.json and configs/other/server-environment-startup.json
-				// Make the config public
-				app.use("/configs", express.static("./configs", options));
-
-				cb();
+			if (ProxyMap.length) {
```

Comment:
I feel like the application of the proxy map belongs outside of the extension. The developer should be able to add to the proxyMap, but it should be processed in code that we control.